internal/command/views/hook_ui.go

```diff
@@ -342,19 +342,67 @@ func (h *UiHook) PrePlanImport(id terraform.HookResourceIdentity, importTarget c
 }
 
 func (h *UiHook) PreApplyImport(id terraform.HookResourceIdentity, importing plans.ImportingSrc) (terraform.HookAction, error) {
-	h.println(fmt.Sprintf(
-		h.view.colorize.Color("[reset][bold]%s: Importing... [id=%s]"),
-		id.Addr, importing.ID,
-	))
+	if importing.Identity != nil {
+		ty, err := importing.Identity.ImpliedType()
+		if err != nil {
+			h.println(fmt.Sprintf(
+				h.view.colorize.Color("[reset][bold]%s: Importing... [identity=(type error)]"),
```

Comment:
I appreciate we don't want this error to necessarily be interruptive but could we at least log it, so that if users do encounter it somehow we have ability to debug it?

Reply:
Do you think the error should be part of the output or are you suggesting another way to log it?

Reply:
I just meant we should log it such that we can then ask the user to set `TF_LOG=DEBUG` and we'd see the actual error in the log. I suppose it means we have to set up the logger somehow somewhere and then call `.Debug()` on it with the whole error.

The only places we log stuff inside of that package currently are two files and this one might be relevant:

https://github.com/hashicorp/terraform/blob/5cc7479e0ec296fc8abf7dfa632597e888f35cbd/internal/command/views/json_view.go#L23-L27

Though there we use it to actually print out the output, so there would be difference in how we set up the logger.

Reply:
👍 I've added `log.Debug` statements for all 4 cases. PTAL